### PR TITLE
Side Panel component

### PR DESCRIPTION
**Pull request by DrShpongle**
Contains:
- side panel
- side panel toggler
- reworked iconic font in Player package

Screen Recording 2021-11-29 at 10 20 34 PM

There are farther improvements (CSS, autoscroll notes list etc) on the plate, will be pushed with next PR.

some thoughts: https://groovy-cement-2a2.notion.site/cue-react-player-805cbf2348064661a61589cf98d405ae

giphy

### CODE REVIEW

File: apps/next-product-starter/src/pages/player.tsx [hunk 1]
```diff
@@ -35,50 +39,108 @@ const PlayerPage = () => {
     videos[0],
   )
 
+  const videoService = useVideo()
+  const withSidePanel = useSelector(videoService, selectWithSidePanel)
+
+  const fullscreenWrapperRef = React.useRef<HTMLDivElement>(null)
+
+  const [isMounted, setMounted] = React.useState<boolean>(false)
```
Comment (DrShpongle) [inline apps/next-product-starter/src/pages/player.tsx:47]: have to be sure Player gets correct ref to the wrapper

File: apps/next-product-starter/src/pages/player.tsx [hunk 2]
```diff
@@ -35,50 +39,103 @@ const PlayerPage = () => {
...
-            <track
-              src={currentVideo.subtitlesUrl}
-              kind="subtitles"
-              srcLang="en"
-              label="English"
-            />
+    <>
+      <div
+        className="relative grid w-full grid-cols-12 gap-0 mx-auto video-with-sidepanel-holder"
+        ref={fullscreenWrapperRef}
```
Comment (DrShpongle) [inline apps/next-product-starter/src/pages/player.tsx:51]: Pass the wrapper as target for the fullscreen

File: packages/player/src/components/controls/side-panel-toggle-control.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  className?: string
+}
+
+export const SidePanelToggleControl: React.FC<SidePanelToggleProps> = ({
+  className,
+}) => {
+  const videoService = useVideo()
+  const withSidePanel = useSelector(videoService, selectWithSidePanel)
+  function handleClick() {
+    videoService.send({type: 'TOGGLE_SIDE_PANEL'})
```
Comment (DrShpongle) [inline packages/player/src/components/controls/side-panel-toggle-control.tsx:17]: action for side panel toggling

File: packages/player/src/components/shortcut.tsx [hunk 4]
```diff
@@ -74,6 +74,15 @@ export const Shortcut: React.FC<ShortcutProps> = ({
...
 
+  React.useEffect(() => {
+    document.addEventListener('fullscreenchange', (e) => {
+      const isFullscreenOff = false === !!document.fullscreenElement
+      if (isFullscreenOff) {
+        videoService.send({type: 'EXIT_FULLSCREEN', element: rootElem})
+      }
+    })
+  }, [])
+
```
Comment (DrShpongle) [inline packages/player/src/components/shortcut.tsx:85]: the code above is needed to listen the exit from fullscreen by clicking on escape button and fire related handler. Without that the state didn't get toggled.

File: packages/player/src/components/side-panel.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  resourceList?: React.ReactNode
+  videoCuesList?: React.ReactNode
+}
+
+export const SidePanel: React.FC<SidePanelProps> = ({
+  resourceList,
+  videoCuesList,
+}) => {
+  return (
+    <div className="relative w-full h-full border border-gray-800 border-l-0 cueplayer-react-side-panel">
```
Comment (joelhooks) [inline packages/player/src/components/side-panel.tsx:14]: these will migrate away from tailwind?

**Review by joelhooks — APPROVED** [review]

File: packages/player/src/components/side-panel.tsx (hunk 5 above)
Comment (DrShpongle) [reply packages/player/src/components/side-panel.tsx:14]: Yes, it will

**Comment by DrShpongle** [comment]
closed in favor of https://github.com/skillrecordings/products/pull/138

**Pull request closed without merging**